subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -659,6 +663,16 @@ int nrf_cloud_decode_data_endpoint(const struct nrf_cloud_data *input,
 		return err;
 	}
 
+	/* Populate batch endpoint topic by copying and appending /batch to the parsed
```

Comment:
perhaps move this section into `nct_dc_endpoint_set`.
then you wouldn't have to add the batch endpoint parameter to the other functions.
i don't think it is needed in `nct_dc_endpoint_get` for example.

Reply:
I did this mostly to align with the other topics. Since `nct_dc_endpoint_get` and `nct_dc_endpoint_set` are both publicly exposed APIs I didn't want to deprive users of not being able to get a topic when the same topic could be set. Its probably unnecessary for the common case though:woman_shrugging:

Reply:
do we consider them "publicly exposed"?  sure, they are not static, but they are in an include within the nrf_cloud lib, not in nrf/include.

perhaps update `nct_dc_endpoint_get` to allow for a NULL batch parameter?
then you could just pass NULL when the function is called in `nrf_cloud_encode_state`.

Reply:
Not directly exposed in that sense. But some users might want to reuse the `nrf_cloud_codec` / expose its API publicly for their application. But my imagination might be running away from me here :milky_way:
I'm happy to update the PR to your suggestion if you think this doesn't make much sense

I'll update `nct_dc_endpoint_get` to allow NULL for the batch_parameter.